addons/project_timesheet_holidays/models/hr_holidays.py

```diff
@@ -55,6 +55,36 @@ class Holidays(models.Model):
 
     timesheet_ids = fields.One2many('account.analytic.line', 'holiday_id', string="Analytic Lines")
 
+    @api.multi
+    def _create_timesheet_lines(self, work_hours_data):
+        self.ensure_one()
+
+        work_hours_data = self.employee_id.list_work_time_per_day(
+            self.date_from,
+            self.date_to,
+        )
+
+        for index, (day_date, work_hours_count) in enumerate(work_hours_data):
+            self.env['account.analytic.line'].create(
+                self._create_timesheet_line_data(index, work_hours_data, day_date, work_hours_count)
+            )
+
+    @api.multi
+    def _create_timesheet_line_data(self, index, work_hours_data, day_date, work_hours_count):
```

Comment:
This is the hook I'm mentioning.

Reply:
Ah, got it. Well, I can move it back, though other modules in Odoo use same approach.

Reply:
Let me rephrase - if this going to be the reason of rejection - I'll move it back, since it was added w/o any particular usage in mind at least right now

Reply:
Yes, as said, 1 change each time.

Reply:
Done, now there's #30867